### ISSUE TITLE
Make watchguard script to set the `engine_signer` on the second validator's node when the first node goes offline

### ISSUE THREAD

**Issue by varasev** [poanetwork/posdao-test-setup#39]
Actually, we can't run two or more nodes by the same validator (with the same `engine_signer`) in AuRa: https://github.com/paritytech/parity-ethereum/issues/10483

As a solution for having the second reserve validator's node, we can launch the second node without `engine_signer` and make a watchguard script which would catch the case when the first node goes offline, and right after that the script would use the `parity_setEngineSigner` RPC call for the second node to make it continue producing the blocks: https://github.com/paritytech/parity-ethereum/issues/10483#issuecomment-473805295

If the first node goes online again, we need to remove the `engine_signer` from the second node.

Let's remove the nodes 4 and 5 from the test setup and leave the node 6 without the `engine_signer` but with such a watchguard script, and test the case when the node 3 goes offline.

**Comment by vkomenda** [comment]
With regards to the suggested AuRa solution, do we really need to be checking whether the signer signs recent blocks? It looks to be from the issue description that we only check whether the primary node is online or not, which is not achieved by looking at the blocks. The same signer will sign blocks when we enable the sealer on the secondary node. So, looking at the blocks does not reveal which of the nodes - the primary or the secondary - has sealed which block with the same block author.

**Comment by varasev** [comment]
I think maybe we need to have two watcher scripts (for both nodes): the script for the first node would work on the first node's machine, the script for the second node - on the second node's machine.

The first script would control somehow that the first node is working fine. The second script would ask the first script whether the first node is OK (through HTTP or socket connection).

Or there can be some other methods of how to determine if the node works fine and is connected to the network: https://wiki.parity.io/JSONRPC-parity-module.html

**Comment by vkomenda** [comment]
I use [isListening](https://web3js.readthedocs.io/en/1.0/web3-eth-net.html#islistening) to check whether the node is online. I imagine there might be even a third machine that monitors the primary and secondary nodes.

**Comment by phahulin** [comment]
For security reasons, it might be best to keep validator node's rpc closed to the external world

**Comment by vkomenda** [comment]
Then RPC calls must be made locally on the secondary node.

**Comment by phahulin** [comment]
Right, I was thinking if `isListening` would work in this case or not, because secondary can't connect to primary's RPC.

**Comment by varasev** [comment]
> I imagine there might even be a third machine that monitors the primary and secondary nodes.

Three machines for each validator is too much, so we should solve this with only two node servers communicating with each other.

**Comment by vkomenda** [comment]
Can you elaborate on the job of the script on the primary node? I think that in our setup we only need one script - imagine it physically runs on the secondary machine. The script checks whether the primary node is running fine using `isListening`. Do we really need to split this function from the single script knowing that the test setup is local?

**Comment by phahulin** [comment]
I think we should write a script that will be used in production too, not only for the test setup, right @varasev ?

**Comment by varasev** [comment]
> I think we should write a script that will be used in production too, not only for the test setup, right @varasev ?

Yes, of course, this task is for production too.
